src/components/MapComponent.vue

```diff
@@ -128,74 +87,32 @@ export default {
     LTooltip,
     LPolyline,
     LControlZoom,
-    LControlAttribution,
     LPopup,
   },
   methods: {
-    setCreateModeTrue() {
-      this.onCreateMode = true;
-      this.locations = [];
-    },
-    check() {
-      console.log("Moved");
-    },
     zoomUpperBound() {
-      if (this.zoom >= 18) {
-        this.zoom = 18;
+      if (this.mapZoom >= 18) {
+        this.$store.commit("setZoom", 18);
       }
     },
     getCurrLoc(e) {
       if (this.onCreateMode === true) {
         try {
           const { lat, lng } = e.latlng;
-          this.locations = [...this.locations, [lat, lng]];
+          // this.locations = [...this.locations, [lat, lng]];
+          this.$store.commit("setLocations", [...this.locations, [lat, lng]]);
+          this.$store.commit("setMarkerVisibility", true);
         } catch (err) {
           console.warn(
             "It's working fine but there are some problems to be fixed" //Its a bug in vue/leaflet itself according to it's github page
           );
         }
       }
     },
-    async createPolyline() {
-      try {
-        //Code
-        if (this.locations.length > 1) {
-          const data = this.$apollo.mutate({
-            mutation: gql`
-              mutation setPath($name: String!, $path: jsonb!) {
-                insert_paths(objects: { path_name: $name, path: $path }) {
-                  affected_rows
-                  returning {
-                    path_name
-                    path
-                  }
-                }
-              }
-            `,
-            variables: {
-              name: this.pathName,
-              path: { latLng: this.locations },
-            },
-          });
-          console.log(data);
-          //Reset Inputs
-          this.pathName = "";
-          this.onCreateMode = false;
-          this.locations = [];
-        } else {
-          alert("Please enter a valid path");
-        }
-      } catch (err) {
-        console.log(err);
-      }
-    },
-    cancel() {
-      this.pathName = "";
-      this.onCreateMode = false;
-      this.locations = [];
-    },
   },
   created() {
+    this.$store.commit("setMarkerVisibility", false);
+    this.$store.commit("setLocations", []);
```

Comment:
@JatinThakur14 Again, these should probably come with default state when the app loads?